proto/info/edge_node_cluster.proto

```diff
@@ -0,0 +1,237 @@
+  STORAGE_VOLUME_PVC_STATUS_AVAILABLE = 3;
+  STORAGE_VOLUME_PVC_STATUS_RELEASED = 4;
+  STORAGE_VOLUME_PVC_STATUS_FAILED = 5;
+}
+
+enum StorageVolumeReplicaStatus {
+  STORAGE_VOLUME_REPLICA_STATUS_UNKNOWN = 0;
+  STORAGE_VOLUME_REPLICA_STATUS_REBUILDING = 1;
+  STORAGE_VOLUME_REPLICA_STATUS_ONLINE = 2;
+  STORAGE_VOLUME_REPLICA_STATUS_FAILED = 3;
+}
+
+message KubePodNameSpaceInfo {
+  // Name of the namespace
+  string name = 1;
+
+  // Number of pods in the namespace
+  uint32 pod_count = 2;
+
+  // Number of pods in the namespace that are running
+  uint32 pod_running_count = 3;
+
+  // Number of pods in the namespace that are pending
+  uint32 pod_pending_count = 4;
+
+  // Number of pods in the namespace that are failed
+  uint32 pod_failed_count = 5;
+
+  // Number of pods in the namespace that are succeeded
+  uint32 pod_succeeded_count = 6;
+}
+
+enum KubePodStatus {
+  KUBE_POD_STATUS_UNKNOWN = 0;
+  KUBE_POD_STATUS_PENDING = 1;
+  KUBE_POD_STATUS_RUNNING = 2;
+  KUBE_POD_STATUS_SUCCEEDED = 3;
+  KUBE_POD_STATUS_CONTAINER_CREATING = 4;
+  KUBE_POD_STATUS_CRASHLOOP_BACKOFF = 5;
+  KUBE_POD_STATUS_ERROR = 6;
+  KUBE_POD_STATUS_EVICTED = 7;
+  KUBE_POD_STATUS_FAILED = 8;
+}
+
+message KubeEVEAppPodInfo {
+  // Name of the EVE application
+  string name = 1;
+
+  // Application Status
+  KubePodStatus status = 2;
+
+  // Restart count of the application
+  uint32 restart_count = 3;
+
+  // Restart time of the application, seconds ago
+  google.protobuf.Timestamp restart_time = 4;
+
+  // Creation Time of the application
+  google.protobuf.Timestamp creation_timestamp = 5;
+
+  // IP address of the application, on cni0 interface
+  string ip_address = 6;
+
+  // Node name on which the application is running
+  string node_name = 7;
+}
+
+message KubeVolumeReplicaInfo {
+  // Name of the volume replica
+  string name = 1;
+
+  // Rebuild progress of the volume replica
+  uint32 rebuild_progress_percentage = 2;
+
+  // Replica status
+  StorageVolumeReplicaStatus status = 3;
+}
+
+message KubeVolumeInfo {
+  // Name of the volume
+  string name = 1;
+
+  // Status of the volume
+  StorageVolumeState state = 2;
+
+  // Robustness of the volume
+  StorageVolumeRobustness robustness = 3;
+
+  // Creation Time of the volume in the cluster
+  google.protobuf.Timestamp creation_timestamp = 4;
+
+  // Provisioned size of the volume in bytes
+  uint64 provisioned_bytes = 5;
+
+  // Allocated size of the volume in bytes
+  uint64 allocated_bytes = 6;
+
+  // PV/PVC status of the volume
+  StorageVolumePVCStatus pvc_status = 7;
+
+  // Replicas of the volume
+  repeated KubeVolumeReplicaInfo replica = 8;
+}
+
+message KubeStorageInfo {
+  // Overall status of Longhorn
+  StorageHealthStatus health = 1;
+
+  // time of the health status transition
+  google.protobuf.Timestamp transition_time = 2;
+
+  // Status of all the volumes in Longhorn
+  repeated KubeVolumeInfo volumes = 3;
+}
+
+enum KubeCompUpgradeStatus {
```

Comment:
What does "comp" mean here and in the next enum?

Reply:
Comp is an installed component in the cluster after first boot completes for eve, some of these are installed once on each node (k3s/containerd) vs others which are installed once per cluster (multus, longhorn, ...).

Reply:
It would make sense to add comments before each enum and message which would spell this out; KubeCompUpgradeStatus enumerates thepossible  status for the kubernetes components, or something in that vein.